### PR TITLE
Initial videohub_client integration

### PR DESCRIPTION
**Pull request by mparent61**
Merging videohub_client.VideohubVideo support into Bulbs for standard Video interface

Each property's Video content models will inherit from new `VideoMixin`

Does not contain any breaking changes, as the updated Video filters aren't yet used by any properties.

### CODE REVIEW

**Comment by mparent61** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;9vgYCmAF&#x2F;33-add-recent-videos-method-with-optional-number-in-content-model"> Add recent videos method with optional number in content model</a>

**Comment by mparent61** [comment]
@benghaziboy @MichaelButkovic 

WIP but want early feedback from the Django Boys. This would start the process of bubbling Video fields into `django-bulbs` via a `VideoMixin` and initial use case of a `recent_videos` search method. It will require standardizing each property's video fields. 

This PR adds `videohub-client` as a dependency, but it's a bit messy since `videohub-client` isn't on PyPI (via **pip** `--process-dependency-links` flag). I'm in favor of eventually merging `videohub_client` into `django-bulbs` entirely, should we just do this now?

**Comment by MichaelButkovic** [comment]
I think I would be in favor of just moving the client in now. Looks good thus far, otherwise.

File: bulbs/content/managers.py [hunk 1]
```diff
@@ -33,10 +33,18 @@ def evergreen(self, included_channel_ids=None, excluded_channel_ids=None, **kwar
     def evergreen_video(self, **kwargs):
         """Filter evergreen content to exclusively video content."""
         eqs = self.evergreen(**kwargs)
-        video_doc_type = getattr(settings, "VIDEO_DOC_TYPE", "")
-        eqs = eqs.filter(es_filter.Type(value=video_doc_type))
+        eqs = eqs.filter(VideohubVideo())
         return eqs
 
+    def recent_videos(self, **kw):
```
Comment (benghaziboy) [inline bulbs/content/managers.py]: 8-)

**Comment by mparent61** [comment]
Now folds entire `videohub-client` library into `videos` app. 

This can be deployed in parallel to videohub_client, but will eventually require client migrations.

**Comment by mparent61** [comment]
@benghaziboy @MichaelButkovic @collin 

I went ahead and merged `videohub-client` into `django-bulbs`. These changes won't break existing properties (but properties will require their own migration dance). When we're done we'll have a more-consistent Video model.

File: setup.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
     "beautifulsoup4>=4.4.1",
     "celery==3.1.10",
     "contextdecorator==0.10.0",
-    "django-betty-cropper>=0.2.0",
+    "django-betty-cropper>=0.2.6",
```
Comment (mparent61) [inline setup.py]: Misc dependency fix for exisiting installations

File: bulbs/videos/models.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
+try:
+    import json
+except ImportError:
```
Comment (benghaziboy) [inline bulbs/videos/models.py]: is this a 2/3 issue?

File: tests/videos/test_videohub_video.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+        video_1 = VideohubVideo.objects.create(title=u"\u2019The Facts Of Life\u2019")
+        video_2 = VideohubVideo.objects.create(title=u"‘The Facts Of Life’")
+
+        # TODO: check this test—is this really a concern?
+        # video_3 = VideohubVideo.objects.create(title="‘The Facts Of Life’")
+
+        self.assertEqual(video_1.get_hub_url(), VideohubVideoSerializer(video_1).data["hub_url"])
+        self.assertEqual(video_2.get_hub_url(), VideohubVideoSerializer(video_2).data["hub_url"])
+        # self.assertEqual(video_3.get_hub_url(), VideohubVideoSerializer(video_3).data["hub_url"])
```
Comment (benghaziboy) [inline tests/videos/test_videohub_video.py]: whysis commented out?

File: bulbs/videos/models.py (hunk 3 above)
Comment (mparent61) [inline bulbs/videos/models.py]: Legacy VHC stuff, removed. Good catch.

File: tests/videos/test_videohub_video.py (hunk 4 above)
Comment (mparent61) [inline tests/videos/test_videohub_video.py]: TERMINATED

File: bulbs/videos/models.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+from djbetty.fields import ImageField
+
+import requests
+
+import six
+
+
+class VideohubVideo(Indexable):
+    """A reference to a video on the onion videohub."""
+    id = models.IntegerField(primary_key=True)
```
Comment (MichaelButkovic) [inline bulbs/videos/models.py]: nitpicky, but django will provide the `id` field here by default

Comment (mparent61) [inline bulbs/videos/models.py]: Good stuff, will update.

**Comment by benghaziboy** [comment]
pending tests 👍 ayyyy

**Comment by MichaelButkovic** [comment]
LGTM👍

**Pull request merged**